packages/kbn-optimizer/src/optimizer/get_output_stats.ts

```diff
@@ -0,0 +1,124 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import Fs from 'fs';
+import Path from 'path';
+
+import { ToolingLog, CiStatsMetrics } from '@kbn/dev-utils';
+import { OptimizerConfig } from './optimizer_config';
+
+const flatten = <T>(arr: Array<T | T[]>): T[] =>
+  arr.reduce((acc: T[], item) => acc.concat(item), []);
+
+interface Entry {
+  relPath: string;
+  stats: Fs.Stats;
+}
+
+const IGNORED_EXTNAME = ['.map', '.br', '.gz'];
+
+const getFiles = (dir: string, parent?: string) =>
+  flatten(
+    Fs.readdirSync(dir).map((name): Entry | Entry[] => {
+      const absPath = Path.join(dir, name);
+      const relPath = parent ? Path.join(parent, name) : name;
+      const stats = Fs.statSync(absPath);
+
+      if (stats.isDirectory()) {
+        return getFiles(absPath, relPath);
+      }
+
+      return {
+        relPath,
+        stats,
+      };
+    })
+  ).filter((file) => {
+    const filename = Path.basename(file.relPath);
+    if (filename.startsWith('.')) {
+      return false;
+    }
+
+    const ext = Path.extname(filename);
+    if (IGNORED_EXTNAME.includes(ext)) {
```

Comment:
Would it be better to specifically include js instead of maintain an ignore list?

Reply:
The files aren't just js files, they include images, fonts, etc, so I think the ignore list is more appropriate.